.travis.yml

```diff
@@ -1,13 +1,26 @@
+services:
+- docker
 script:
-- docker build --target test --tag todo-app:test .
-- docker run todo-app:test tests/test_viewmodel.py
-- docker run --env-file .env.test todo-app:test tests/test_app.py
+- docker build --target development --tag vaibhav1s/todo-app-dev:${TRAVIS_COMMIT} .
+- docker build --target test --tag vaibhav1s/todo-app-test:${TRAVIS_COMMIT} .
+- docker run vaibhav1s/todo-app-test:${TRAVIS_COMMIT} tests/test_viewmodel.py
+- docker run --env-file .env.test vaibhav1s/todo-app-test:${TRAVIS_COMMIT} tests/test_app.py
 - docker run --env TRELLO_BOARD_NAME_CREATE=E2E_test_board --env TRELLO_KEY
-  --env TRELLO_TOKEN todo-app:test tests/test_trello_api_calls.py
+  --env TRELLO_TOKEN vaibhav1s/todo-app-test:${TRAVIS_COMMIT} tests/test_trello_api_calls.py
 - docker run --env TRELLO_BOARD_NAME_CREATE=E2E_test_board --env TRELLO_USER_NAME=vaibhavsharma206
   --env TRELLO_BOARD_NAME=team_template --env TRELLO_KEY --env TRELLO_TOKEN
   --env SECRET_KEY
-  todo-app:test tests_e2e/test_e2e.py
+  vaibhav1s/todo-app-test:${TRAVIS_COMMIT} tests_e2e/test_e2e.py
+after_success: 
+- docker build --target production --tag vaibhav1s/todo-app-prod:${TRAVIS_COMMIT} .
+- echo "$DOCKER_PASSWORD" | docker login --username vaibhav1s --password-stdin
+- docker push vaibhav1s/todo-app-prod:${TRAVIS_COMMIT}
+- docker tag vaibhav1s/todo-app-prod:${TRAVIS_COMMIT} registry.heroku.com/todo-app-devops/web
+- docker login --username=_ --password=${HEROKU_AUTH_TOKEN} registry.heroku.com
+- docker push registry.heroku.com/todo-app-devops/web
+- heroku container:login
```

Comment:
This is equivalent to your `docker login ... registry.heroku.com` command. It logs into the image registry.
No need for both.

Reply:
I have corrected this.